### ISSUE TITLE
CXF Host issue when migrating from Wildfly 8.2.0 to Wildly 9.0.0cr1

### ISSUE THREAD

**Issue by ugol** [wildfly-extras/wildfly-camel#804]
We tried to migrate a camel-cxf project from Wildfly 8.2.0 (CXF 2.7.something) to Wildly 9.0.0cr1 (CXF 3.0.4). The code is straightforward and is based on one of the provided examples, it's a simple camel route that at the end just calls an axis 2 web server (exposed by Websphere).

The problem is that trying it on the new wildfly-camel 3.0.0 the route gets a 404 error, and apparently this is due to a wrong "Host" header in the soap request containing "127.0.0.1".
Brutally removing the header with Camel works, but I think there is a bug somewhere, probably CXF 3 treats the Host header differently?

**Comment by jamesnetherton** [comment]
I've done a very quick test where I deployed a [basic web service](https://github.com/wildfly-extras/wildfly-camel/blob/master/examples/camel-cxf/src/main/java/org/wildfly/camel/examples/cxf/GreetingServiceImpl.java) to a remote host running WildFly 9.0.0.CR1.

I then used wildfly-camel 3.0.0 to run a simple CXF Camel route on my local machine to consume the remote service:

```
from("direct:start")
.to("cxf://http://thanos:8080/example-camel-cxf/greeting?serviceClass=" + GreetingService.class.getName());
```

It all seems to be working fine. If I sniff the HTTP traffic being sent on POST to the web service, the headers all look good. My remote host is named 'thanos' and the 'Host' header is named correctly.

![CXF Request](http://snag.gy/olghq.jpg)

What does your Camel route look like? Maybe something is affecting the host header value.
